frontend/src/metabase/dashboard/actions/tabs.ts

```diff
@@ -81,200 +88,171 @@ export function getDefaultTab({
   };
 }
 
-export const tabsReducer = handleActions<DashboardState, TabsReducerPayload>(
-  {
-    [CREATE_NEW_TAB]: (
-      state,
-      { payload: { tabId } }: Action<CreateNewTabPayload>,
-    ) => {
-      const { dashId, prevDash, prevTabs } = getPrevDashAndTabs(state);
-      if (!dashId || !prevDash) {
-        throw Error(
-          `CREATE_NEW_TAB was dispatched but either dashId (${dashId}) or prevDash (${prevDash}) are null`,
-        );
-      }
-
-      // Case 1: Dashboard already has tabs
-      if (prevTabs.length !== 0) {
-        // 1. Create new tab, add to dashboard
-        const newTab = getDefaultTab({
-          tabId,
-          dashId,
-          name: t`Page ${prevTabs.length + 1}`,
-        });
-        const dashboards: DashboardState["dashboards"] = {
-          ...state.dashboards,
-          [dashId]: {
-            ...prevDash,
-            ordered_tabs: [...prevTabs, newTab],
-          },
-        };
-
-        // 2. Select new tab
-        const selectedTabId = tabId;
-
-        return { ...state, dashboards, selectedTabId };
-      }
-
-      // Case 2: Dashboard doesn't have tabs
-
-      // 1. Create two new tabs, add to dashboard
-      const firstTabId = tabId + 1;
-      const secondTabId = tabId;
-      const newTabs = [
-        getDefaultTab({ tabId: firstTabId, dashId, name: t`Page 1` }),
-        getDefaultTab({ tabId: secondTabId, dashId, name: t`Page 2` }),
-      ];
-      const dashboards: DashboardState["dashboards"] = {
-        ...state.dashboards,
-        [dashId]: {
-          ...prevDash,
-          ordered_tabs: [...prevTabs, ...newTabs],
-        },
-      };
-
-      // 2. Select second tab
-      const selectedTabId = secondTabId;
-
-      // 3. Assign existing dashcards to first tab
-      const dashcards: DashboardState["dashcards"] = { ...state.dashcards };
-      if (prevTabs.length === 0) {
+export const tabsReducer = createReducer<DashboardState>(
```

Comment:
I went ahead and refactored all cases for this reducer to use RTK's mutable immer draft state. If you want to review the new undo functionality, see the two cases `deleteTab` and `undoDeleteTab`